components/kyma-metrics-collector/pkg/gardener/shoot/client.go

```diff
@@ -32,12 +33,17 @@ func NewClient(opts *options.Options) (*Client, error) {
 
 func (c Client) Get(ctx context.Context, shootName string) (*gardenerv1beta1.Shoot, error) {
 	unstructuredShoot, err := c.ResourceClient.Get(ctx, shootName, metaV1.GetOptions{})
-	if err != nil {
+
+	if err == nil {
+		gardenercommons.TotalCalls.WithLabelValues(gardenercommons.SuccessStatusLabel, shootName, gardenercommons.SuccessGettingShootLabel).Inc()
+		return convertRuntimeObjToShoot(unstructuredShoot)
+	}
+
+	if !errors.IsNotFound(err) {
 		gardenercommons.TotalCalls.WithLabelValues(gardenercommons.FailureStatusLabel, shootName, gardenercommons.FailedGettingShootLabel).Inc()
-		return nil, err
 	}
```

Comment:
Should we log an error message if it's a NotFound error?

Reply:
It's already logged one level above (see: https://github.com/kyma-project/control-plane/blob/2093b3b930baada408b5622ca8d801f5201a3e06/components/kyma-metrics-collector/pkg/process/process.go#L98)